src/mon/OSDMonitor.cc

```diff
@@ -942,6 +942,18 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   }
   mapping_job.reset();
 
+  // ensure we don't have blank new_state updates.  these are interrpeted as
+  // CEPH_OSD_UP (and almost certainly not what we want!).
+  auto p = pending_inc.new_state.begin();
+  while (p != pending_inc.new_state.end()) {
+    if (p->second == 0) {
```

Comment:
@liewegas but how come can we add a `0` new_state mask? this happens only if we have a new `0` state mask in the incoming MOSDFull. the only case i can think of is #16581.